### PR TITLE
add json context to metrics and latest_metrics

### LINKED ISSUE #580: Metric contexts: adapt the database schema
At the very least, a new column with JSON type needs to be added to the `metrics` and `latest_metrics` tables. Ideally we would normalize the metrics IDs by having a new `metrics` table that maps a `key` and `context` to an `id`, and then reference those IDs in `metric_values` and `latest_metric_values` (new table names) instead of the `key`.

### PR DESCRIPTION
**Pull request by suprjinx**
This PR adds database support for json context columns in the `metrics` and `latest_metrics` tables.

Fixes #580

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -39,6 +39,11 @@ require (
 	gorm.io/plugin/dbresolver v1.4.7
 )
 
+require (
+	github.com/go-sql-driver/mysql v1.7.0 // indirect
+	gorm.io/driver/mysql v1.4.7 // indirect
+)
+
```
Comment (dsuhinin) [inline go.mod:46]: was that introduced by `datatypes` package?

Comment (suprjinx) [reply go.mod:46]: yes

File: pkg/api/mlflow/dao/models/metric.go [hunk 2]
```diff
@@ -9,6 +11,8 @@ type Metric struct {
 	Step      int64   `gorm:"default:0;not null;primaryKey"`
 	IsNan     bool    `gorm:"default:false;not null;primaryKey"`
 	Iter      int64   `gorm:"index"`
+	ContextID *uint
```
Comment (dsuhinin) [inline pkg/api/mlflow/dao/models/metric.go:14]: maybe we have to use instead `sql.Null` type?

Comment (suprjinx) [reply pkg/api/mlflow/dao/models/metric.go:14]: i think pointer type is implicitly nullable, yes?

Comment (dsuhinin) [reply pkg/api/mlflow/dao/models/metric.go:14]: I think we have to follow our existing principles -> https://github.com/G-Research/fasttrackml/blob/main/pkg/database/migrations/v_0006/model.go#L50 for example.

Comment (dsuhinin) [reply pkg/api/mlflow/dao/models/metric.go:14]: in most of the cases we use `sql.Null` types.

Comment (suprjinx) [reply pkg/api/mlflow/dao/models/metric.go:14]: using pointer for now as with other recent additions

**Review by dsuhinin — APPROVED** [review]
looks good!

**Pull request merged**